core/proto/core.proto

```diff
@@ -40,4 +39,30 @@ message Certificate {
   optional int64 expires = 6; // Unix timestamp (nanoseconds)
 }
 
+message Registration {
+        optional int64 id = 1;
+        optional bytes key = 2;
+        repeated string contact = 3;
+        optional bool contactsPresent = 4;
+        optional string agreement = 5;
+        optional bytes initialIP = 6;
+        optional int64 createdAt = 7; // Unix timestamp (nanoseconds)
+        optional string status = 8;
+}
+
+message Authorization {
+        optional string id = 1;
+        optional string identifier = 2;
+        optional int64 registrationID = 3;
+        optional string status = 4;
+        optional int64 expires = 5; // Unix timestamp (nanoseconds)
+        repeated core.Challenge challenges = 6;
+        optional bytes combinations = 7;
+}
+
+message CertificateRequest {
```

Comment:
I think there's a little confusion here. I see now that `CertificateRequest` is a type that's already defined in `core/objects.go`. I'm fine with defining PB equivalents to the `core/objects.go` types - that's the right direction. However, the `message CertificateRequest` you define here is not the same as the `type CertificateRequest` defined in `core/objects.go`. Here we have:
```
        optional bytes csr = 1;
        optional int64 regID = 2;
```

But in `core/objects.go` we have:

```
type CertificateRequest struct {
	CSR   *x509.CertificateRequest // The CSR
	Bytes []byte                   // The original bytes of the CSR, for logging.
}
```

I think that difference alone is enough to avoid doing this.

Instead, I think `ra.proto` should define a `message NewCertificateRequest` to go with `rpc NewCertificate`, and `sa.proto` should define a `message AddCertificateRequest` to go with `rpc AddCertificate`. There will be some small amount of code in `grpc-wrappers.go` that could be de-duplicated, but keeping a consistent pattern is worth much more than that. I feel like I read some commentary by the Go authors about, "syntactic duplication is not as bad as semantic duplication."